src/stores/ResourceConfig.ts

```diff
@@ -0,0 +1,240 @@
+import { ResourceConfigStoreNames } from 'context/Zustand';
+import { LiveSpecsExtQuery } from 'hooks/useLiveSpecsExt';
+import produce from 'immer';
+import { difference, has, isEmpty, isEqual, map, omit } from 'lodash';
+import { createJSONFormDefaults } from 'services/ajv';
+import { JsonFormsData, Schema } from 'types';
+import { devtoolsOptions } from 'utils/store-utils';
+import create, { StoreApi } from 'zustand';
+import { devtools, NamedSet } from 'zustand/middleware';
+
+// TODO: Narrow as many types as possible.
+export interface ResourceConfig {
+    [key: string]: JsonFormsData | any[];
+    errors: any[];
+}
+
+interface ResourceConfigDictionary {
+    [key: string]: ResourceConfig;
+}
+
+// TODO: Determine whether the resourceConfig state property should be made plural. It is a dictionary of individual resource configs, so I am leaning "yes."
+// TODO: Determine if it is possible to break out the collection selector segment of the store. Feels like an anti-pattern of this new approach.
+export interface ResourceConfigState {
+    // Collection Selector
+    collections: string[] | null;
+    preFillCollections: (collections: LiveSpecsExtQuery[]) => void;
+
+    collectionErrorsExist: boolean;
+
+    currentCollection: string | null;
+    setCurrentCollection: (collections: string | null) => void;
+
+    // Resource Config
+    resourceConfig: ResourceConfigDictionary;
+    setResourceConfig: (
+        key: string | [string],
+        resourceConfig?: ResourceConfig
+    ) => void;
+
+    resourceConfigErrorsExist: boolean;
+    resourceConfigErrors: (string | undefined)[];
+
+    // Resource Schema
+    resourceSchema: Schema;
+    setResourceSchema: (val: ResourceConfigState['resourceSchema']) => void;
+
+    // Misc.
+    stateChanged: () => boolean;
+    resetState: () => void;
+}
+
+const populateResourceConfigErrors = (
+    resourceConfig: ResourceConfigDictionary,
+    state: ResourceConfigState
+): void => {
+    let resourceConfigErrors: any[] = [];
+
+    if (Object.keys(resourceConfig).length > 0) {
+        map(resourceConfig, (config) => {
+            const { errors } = config;
+
+            // TODO: Determine whether it is possible for errors to not exist.
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+            if (errors && errors.length > 0) {
+                resourceConfigErrors = resourceConfigErrors.concat(errors);
+            }
+        });
+    } else {
+        // TODO: Get comment clarity from Travis. What is the motivation for the comment below?
+        // TODO (errors) Need to populate this object with something?
```

Comment:
What is the motivation for this comment?

Reply:
Might want to add actual details of the specific error to display to the user. Right now we just check if there is an error and display a general "hey this section isn't right" kind of error.